contracts/AllDay.cdc

```diff
@@ -558,6 +569,269 @@ access(all) contract AllDay: NonFungibleToken {
+            assert(self.slugToBadge[badgeSlug] != nil, message: "badge doesn't exist")
+            
+            // Initialize the dictionary for the momentID's badge slugs if it doesn't exist
+            if self.momentIdToBadgeSlugs[momentID] == nil {
+                self.momentIdToBadgeSlugs[momentID] = {}
+            }
+
+            // Get a reference to the momentID's badge slug
+            // using a reference ensures the whole badge slugs are not loaded to memory
+            let momentBadgeSlugsRef = &self.momentIdToBadgeSlugs[momentID] as auth(Insert) &{String: {String: String}}?
+                ?? panic("Could not get a reference to the moment's badge slugs")
+
+            assert(momentBadgeSlugsRef[badgeSlug] == nil, message: "badge slug already added to moment")
+            
+            // Insert the slug
+            momentBadgeSlugsRef.insert(key: badgeSlug, metadata)
+            
+            emit BadgeAddedToMoment(badgeSlug: badgeSlug, MomentID: momentID, metadata: metadata)
+        }
+
+        // Remove badge from play
+        access(ManageBadges) fun removeBadgeFromPlay(badgeSlug: String, playID: UInt64){
+            if let playBadges = &self.playIdToBadgeSlugs[playID] as auth(Remove) &{String: {String: String}}? {
+                if playBadges.containsKey(badgeSlug) {
+                    playBadges.remove(key: badgeSlug)
+                    emit BadgeRemovedFromPlay(badgeSlug: badgeSlug, playID: playID)
+                }
+            }
+        }
+
+        // Remove badge from edition
+        access(ManageBadges) fun removeBadgeFromEdition(badgeSlug: String, editionID: UInt64){
+            if let editionBadges = &self.editionIdToBadgeSlugs[editionID] as auth(Remove) &{String: {String: String}}? {
+                if editionBadges.containsKey(badgeSlug) {
+                    editionBadges.remove(key: badgeSlug)
+                    emit BadgeRemovedFromEdition(badgeSlug: badgeSlug, EditionID: editionID)
+                }
+            }
+        }
+
+        // Remove badge from moment
+        access(ManageBadges) fun removeBadgeFromMoment(badgeSlug: String, momentID: UInt64){
+            if let momentBadges = &self.momentIdToBadgeSlugs[momentID] as auth(Remove) &{String: {String: String}}? {
+                if momentBadges.containsKey(badgeSlug) {
+                    momentBadges.remove(key: badgeSlug)
+                    emit BadgeRemovedFromMoment(badgeSlug: badgeSlug, MomentID: momentID)
+                }
+            }
+        }
+    }  
+
+    access(contract) view fun getBadgesStoragePath(): StoragePath{
+        return /storage/AllDayAdminBadges
+    }
+
+    access(contract) fun initializeBadgesInStorage(){
+        let path = AllDay.getBadgesStoragePath()
+        let badges <- create Badges()
+        self.account.storage.save(<-badges, to: path)
+    }
+
+    access(contract) fun borrowBadges(): &AllDay.Badges?{
```

Comment:
Nit: use view modifier, same for getBadge

Reply:
doesn't work, because of the storage access